freqtrade/freqtradebot.py

```diff
@@ -543,12 +543,19 @@ def check_and_call_adjust_trade_position(self, trade: Trade):
         min_stake_amount = self.exchange.get_min_pair_stake_amount(trade.pair,
                                                                    current_rate,
                                                                    self.strategy.stoploss)
-        max_stake_amount = self.wallets.get_available_stake_amount()
+        max_stake_amount = self.exchange.get_max_pair_stake_amount(trade.pair,
+                                                                   current_rate,
+                                                                   self.strategy.stoploss)
+        if max_stake_amount is None:
+            # * Should never be executed
+            raise OperationalException(f'max_stake_amount is None for {trade}')
```

Comment:
It's not possible, it's because `Exchange._get_stake_amount_limit` can return `None` for min, but I moved the check to the Exchange class inside `get_max_pair_stake_amount`. It's there to prevent programmer and mypy errors, but I can also just set `# ignore` inside `get_max_pair_stake_amount`